src/hotspot/cpu/riscv/riscv.ad

```diff
@@ -1191,6 +1190,30 @@ int MachCallNativeNode::ret_addr_offset() {
   return -1;
 }
 
+// With RVC a call may get 2-byte aligned.
+//   The offset encoding in jal ranges bits [12, 31] could span the cache line.
+//   Patching this unaligned address will make the write operation not atomic.
+//   Other threads may be running the same piece of code at full speed, causing concurrency issues.
+//   So we must ensure that it does not span a cache line so that it can be patched.
```

Comment:
Suggestion:

// With RVC a call site may get 2-byte aligned.
// The offset encoding in jal instruction bits [12, 31] could span multiple cache lines.
// Patching this jal instruction will not be atomic when its address is not naturally aligned.
// Other threads may be running the same piece of code at the same time, thus causing concurrency issues.